### ISSUE TITLE
CM autoclose tag behavior with 'whenClosing' on typing '/'

### ISSUE THREAD

**Issue by swmitra** [codemirror/CodeMirror#4242]
When autoclose tag is configured with 'whenClosing', if user opens an existing html file which contains some invalid tags and try to close them, cm inserts closing tags with '>' which introduces new syntax errors in the html. The screen capture below depicts the error scenario - 
![animation](https://cloud.githubusercontent.com/assets/12087205/18504646/fd2d49b2-7a80-11e6-9b9d-3d2acb979105.gif)

IMO, the logical behavior could be not to insert any close tag, as the '/' typed itself closed the open tag in the context.

**Comment by marijnh** [comment]
Do you want to implement the behavior you describe?

**Comment by swmitra** [comment]
@marijnh Thanks for the quick response :+1: 
If you think it's a valid behavior what I have proposed, then yes, I will try to implement the new behavior and raise a PR.

**Comment by marijnh** [comment]
Yes, it sounds valid (_when_ the inserted content exactly matches the text after the cursor -- when not, I think it should just do the dumb thing it does now)
